OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -0,0 +1,457 @@
+    SimTK::Array_<double> sqMarkerErrors;
+    double looseSumSqError = 0;
+    ikSolver.computeCurrentSquaredMarkerErrors(sqMarkerErrors);
+    for (const double& err : sqMarkerErrors) {
+        looseSumSqError += err;
+    }
+    
+    cout << "For accuracy: " << looseAccuracy << "; Sum-squared Error: " 
+        << looseSumSqError << endl;
+
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    ikSolver.setAccuracy(tightAccuracy);
+    ikSolver.assemble(state);
+
+    coordValue = coord.getValue(state);
+    cout << "Assembled " << coord.getName() <<" value = "<< coordValue << endl;
+
+    accuracy = abs(coord.getValue(state) - refVal);
+
+    cout << "Specified accuracy: " << tightAccuracy << "; achieved: "
+        << accuracy << endl;
+
+    // test should fail if we did not meet the new tighter target accuracy
+    SimTK_ASSERT_ALWAYS(accuracy <= tightAccuracy,
+        "InverseKinematicsSolver failed to meet specified accuracy");
+
+    double tightSumSqError = 0;
+    ikSolver.computeCurrentSquaredMarkerErrors(sqMarkerErrors);
+    for (const double& err : sqMarkerErrors) {
+        tightSumSqError += err;
+    }
+
+    cout << "For accuracy: " << tightAccuracy << "; Sum-squared Error: "
+        << tightSumSqError << endl;
+
+    // refining the accuracy should not increase tracking errors
+    SimTK_ASSERT_ALWAYS(tightSumSqError <= looseSumSqError,
+        "InverseKinematicsSolver failed to maintain or lower marker errors "
+        "when accuracy was tightened.");
+}
+
+void testUpdateMarkerWeights()
+{
+    cout << "\ntestInverseKinematicsSolver::testUpdateMarkerWeights()" << endl;
+
+    std::unique_ptr<Model> pendulum{ constructPendulumWithMarkers() };
+    Coordinate& coord = pendulum->getCoordinateSet()[0];
+
+    double refVal = 0.123456789;
+
+    SimTK::State state = pendulum->initSystem();
+    coord.setValue(state, refVal);
+
+    StatesTrajectory states;
+    states.append(state);
+
+    MarkerData* markerData =
+        generateMarkerDataFromModelAndStates(*pendulum, states, 0.02);
+    MarkersReference markersRef(markerData);
+    auto& markerNames = markersRef.getNames();
+
+    for (const auto& name : markerNames) {
+        markersRef.updMarkerWeightSet().adoptAndAppend(
+            new MarkerWeight(name, 1.0));
+    }
+
+    SimTK::Array_<CoordinateReference> coordRefs;
+    // Reset the initial coordinate value
+    coord.setValue(state, 0.0);
+    InverseKinematicsSolver ikSolver(*pendulum, markersRef, coordRefs);
+    ikSolver.setAccuracy(1.0e-8);
+    ikSolver.assemble(state);
+
+    double coordValue = coord.getValue(state);
+    cout << "Assembled " << coord.getName() << " value = "
+        << coordValue << endl;
+
+    SimTK::Array_<double> nominalMarkerErrors;
+    ikSolver.computeCurrentMarkerErrors(nominalMarkerErrors);
+
+    SimTK::Array_<double> markerWeights;
+    markersRef.getWeights(state, markerWeights);
+
+    for (unsigned int i = 0; i < markerNames.size(); ++i) {
```

Comment:
Test looks great but uses MarkerData class, will that need to change with pending phasing out of MarkerData class?